lightning/src/routing/gossip.rs

```diff
@@ -888,12 +889,14 @@ impl ChannelInfo {
 				return None;
 			}
 		};
-		direction.map(|dir| (DirectedChannelInfo::new(self, dir, outbound), source))
+		let dir = direction.expect("We checked that both directions are available at the start");
```

Comment:
Also, one quick question.

Usage of `expect` here would create an explicit panic in case `direction.is_none()`.
But would explicitly panicking in a `pub fn` be a good idea?

Should we consider returning a `Result<Ok, Err>`, instead of panicking within the function?

Reply:
Ah! Right!
Then it makes sense! :)